From: David Zeuthen <davidz@redhat.com>
Date: Wed, 4 Oct 2006 20:10:40 +0000 (-0400)
Subject: don't add non-existing partition entries for MS-DOS disk labels
X-Git-Url: http://gitweb.freedesktop.org/?p=hal.git;a=commitdiff;h=51b841cd59a73fc093535ddd9e512d3036ebbbc4

don't add non-existing partition entries for MS-DOS disk labels

I think the issue is that the users who run into this have broken
partition extended tables, e.g. we error out in the function
part_table_probe_msdos_extended() and thus return NULL for the
extended partition table. We then stuff a NULL pointer into the
entries list and that's causing the crash on freeing the partition
table or looking through entries. This patch should fix at least the
crashes.
---

--- a/partutil/partutil.c
+++ b/partutil/partutil.c
@@ -585,7 +585,9 @@ part_table_parse_msdos (int fd, guint64 
 
 		//HAL_INFO (("pe = %p", pe));
 
-		p->entries = g_slist_append (p->entries, pe);
+		if (pe != NULL) {
+			p->entries = g_slist_append (p->entries, pe);
+		}
 	}
 
 out:

